evil-common.el

```diff
@@ -2424,17 +2429,50 @@ The tracked insertion is set to `evil-last-insertion'."
 
 ;;; Paste
 
+(defun evil-set-register-on-yank (register text)
+  "Set REGISTER to TEXT.
+Additionally set register 0 and shift the remaining number
+registers if necessary."
+  (when register
+    (evil-set-register register text))
+  (unless (eq register ?_)
+    (kill-new text)
+    (unless (or register evil-is-yank-and-delete)
+      ;; "0 register contains last yanked text
+      (evil-set-register ?0 text)))
+  ;; Handle 1-9 registers when deleting
+  ;; http://vimdoc.sourceforge.net/htmldoc/change.html#registers
+  (when (and evil-is-yank-and-delete (not (eq register ?_)))
+    (let ((special-delete-motions
```

Comment:
This should be put into a customizable variable, so that other packages can add their own things to it.